.travis.yml

```diff
@@ -13,32 +10,62 @@ before_install:
   - git remote set-branches --add origin master && git fetch
   # fetch full history for correct current and previous version detection
   - git fetch --unshallow
-  - git branch -f "$TRAVIS_BRANCH" && git checkout "$TRAVIS_BRANCH"
   # using jabba for custom jdk management
   - curl -sL https://raw.githubusercontent.com/shyiko/jabba/0.11.1/install.sh | bash && . ~/.jabba/jabba.sh
   - jabba install adopt@~1.8.202-08
   - jabba install adopt@~1.11.0-1
-  - docker-compose up -d cassandra
 
 before_script:
   - unset _JAVA_OPTIONS
 
 jobs:
   include:
+    - stage: check
+      script: sbt scalafmtCheckAll || { echo "[error] Unformatted code found. Please run 'scalafmtAll' and commit the reformatted code."; false; }
+      name: "Code style check (fixed with `sbt scalafmtAll`)"
+    - script: sbt scalafmtSbtCheck || { echo "[error] Unformatted sbt code found. Please run 'scalafmtSbt' and commit the reformatted code."; false; }
+      name: "Build code style check (fixed with `sbt scalafmtSbt`)"
+    - script: sbt ++2.13.0-RC2 Test/compile"
+      name: "Compile all tests (with Scala 2.13)"
+    - script: sbt unidoc
+      name: "Create all API docs"
+    - script: sbt docs/paradox
+      name: "Create site with Paradox"
+
     - stage: test
-      script: jabba use $JDK  && java -version && sbt ++$TRAVIS_SCALA_VERSION ";scalafmtCheckAll;test"
+      script: jabba use adopt@~1.8.202-08 && java -version && docker-compose up -d cassandra && sbt test
```

Comment:
Shouldn't this run tests for all Scala versions? `+test`?

Reply:
Good catch.